src/usb_moded-udev.c

```diff
@@ -421,6 +421,7 @@ static void umudev_parse_properties(struct udev_device *dev, bool initial)
         log_debug("CONNECTED - POWER_SUPPLY_TYPE = %s", power_supply_type);
 
         if( !strcmp(power_supply_type, "USB") ||
+            !strcmp(power_supply_type, "USB_PD") ||
```

Comment:
I'm under the impression that PD implies charger rather than pc being connected.

If that is true, then this ought to be treated like "USB_DCP" & co below (yielding CABLE_STATE_CHARGER_CONNECTED), not like "USB" (yielding CABLE_STATE_PC_CONNECTED).

Alternatively, if PD can apply for both charger and pc connections, is there some way to tell those apart?

Reply:
This is what is reported with a Pixel 3a XL connected to the USB-C port of a Thinkpad P14s Gen 2 running Ubuntu 22.04. Unfortunately, I still cannot get a reliable connection, it works fine when connected to a USB-A port, i.e. without the USB-PD availability. Not sure if it is the cable, phone or notebook.

I need to re-check with another cable and other phones as soon as I get UBports focal with modern usb-moded running on them, so I'm marking this as draft until then.

Reply:
I played around with this for a while. Using XA2, various chargers, ubs hubs, power banks and usb c port in my thinkpad & docking statation...

Short version, AFAICT etc: USB_PD can show up both on charger and pc connections. And there are no property values that can be used to explicitly tell which kind it is in stable state. But there are transient differences at connect time -> I made a PR that takes that into account: https://github.com/sailfishos/usb-moded/pull/9 - would be nice if you can test it out.

Seems to work quite well with the assorted hw combo that I happened to have available. But I did notice that occasionally laptop-phone connection goes into some backwards state and the phone no longer reacts to cable disconnect / connect. When that happens, detaching both ends of the cable seems to resume normal operation.